rust/clients/host_agent/src/main.rs

```diff
@@ -36,28 +36,36 @@ async fn main() -> Result<(), AgentCliError> {
 
     let cli = agent_cli::Root::parse();
     match &cli.scope {
-        Some(agent_cli::CommandScopes::Daemonize) => {
+        agent_cli::CommandScopes::Daemonize(daemonize_args) => {
             log::info!("Spawning host agent.");
-            daemonize().await?;
-        }
-        Some(agent_cli::CommandScopes::Host { command }) => host_cmds::host_command(command)?,
-        Some(agent_cli::CommandScopes::Support { command }) => {
-            support_cmds::support_command(command)?
-        }
-        None => {
-            log::warn!("No arguments given. Spawning host agent.");
-            daemonize().await?;
+            daemonize(daemonize_args).await?;
         }
+        agent_cli::CommandScopes::Host { command } => host_cmds::host_command(command)?,
+        agent_cli::CommandScopes::Support { command } => support_cmds::support_command(command)?,
     }
 
     Ok(())
 }
 
-async fn daemonize() -> Result<(), async_nats::Error> {
+async fn daemonize(args: &DaemonzeArgs) -> Result<(), async_nats::Error> {
     // let (host_pubkey, host_creds_path) = auth::initializer::run().await?;
-    let host_creds_path = nats_js_client::get_nats_client_creds("HOLO", "HPOS", "hpos");
-    let host_pubkey = "host_id_placeholder>";
-    gen_leaf_server::run(&host_creds_path).await;
-    workload_manager::run(host_pubkey, &host_creds_path).await?;
+    let _ = gen_leaf_server::run(
+        &args.nats_leafnode_client_creds_path,
+        &args.store_dir,
+        args.hub_url.clone(),
+        args.hub_tls_insecure,
+    )
+    .await;
+
+    let _ = workload_manager::run(
+        "host_id_placeholder>",
+        &args.nats_leafnode_client_creds_path,
+        args.nats_connect_timeout_secs,
+    )
+    .await?;
+
+    // Only exit program when explicitly requested
+    tokio::signal::ctrl_c().await?;
+
```

Comment:
```suggestion
    let host_client = workload_manager::run(
        "host_id_placeholder>",
        &args.nats_leafnode_client_creds_path,
        args.nats_connect_timeout_secs,
    )
    .await?;

    // Only exit program when explicitly requested
    tokio::signal::ctrl_c().await?;
    
    host_client.close().await?;
```

It looks like we're returning the worload manager nats client back here.  Is there a reason e shouldn't be reading it to close the client after the "exit" request is received?